Add "Branch workflow" (#198)
yiisoft/docs: 006-git.md

# 006 - Git

## Branch workflow

- Create a dedicated branch for each feature. Develop each version in specific branches.
- Before the first release, main branch is `master`.

> - After the first release, create a new release branch `1.x` and make it main branch.

Reviewer: Does it worth making it main branch if we have master and it's "latest"?

Author: master for major changes, other changes should be added to branch with current major version.

> - Major changes must create in `master` branch.

Reviewer: ```suggestion
- Major changes must be made in `master` branch.
```

# 006 - Git

## Branch workflow

> For each feature is used a dedicated branch. The development of each version is carried out in the specific branches.
> 
> Before the first release, main branch is `master` only.
> 
> After the first release, `master` branch must rename to `1.x`. For major change must be created a new branch with usage
> next version as name (for example, `2.x`). The main branch is the branch with the current stable release.

Reviewer: ```suggestion
- Create a dedicated branch for each feature. Develop each version in specific branches.
- Before the first release, main branch is `master`.
- After the first release, rename `master` to `1.x`. For a major release, create a new version branch such as `2.x`. The main branch is the branch with the current stable release.
```
